launcher/launcher.go

```diff
@@ -36,3 +63,805 @@ type NodeInfo struct {
+			node.NodeID, node.currentRunRequest.services, node.currentRunRequest.layers, node.currentRunRequest.instances,
+			true)
+	}
+
+	return nil
+}
+
+// GetInstanceStatsesChanhel gets channel with instances status.
+func (launcher *Launcher) GetInstanceStatsesChanhel() <-chan []cloudprotocol.InstanceStatus {
+	return launcher.runInstacnesStatusChannel
+}
+
+// GetNodesConfiguration gets nodes configuration.
+func (launcher *Launcher) GetNodesConfiguration() []cloudprotocol.NodeInfo {
+	nodes := make([]cloudprotocol.NodeInfo, len(launcher.nodes))
+
+	i := 0
+
+	for _, v := range launcher.nodes {
+		nodes[i] = v.NodeInfo.NodeInfo
+		i++
+	}
+
+	return nodes
+}
+
+/***********************************************************************************************************************
+ * Private
+ **********************************************************************************************************************/
+
+func (launcher *Launcher) fillUIDPool() {
+	uids, err := launcher.storage.GetAllUIDs()
+	if err != nil {
+		log.Errorf("Can't fill UID pool: %v", err)
+	}
+
+	for _, uid := range uids {
+		if err = launcher.uidPool.AddID(uid); err != nil {
+			log.Errorf("Can't add UID to pool: %v", err)
+		}
+	}
+}
+
+func (launcher *Launcher) processChannels(ctx context.Context) {
+	for {
+		select {
+		case instances := <-launcher.nodeManger.GetRunInstancesStatusChannel():
+			launcher.processRunInstanceStatus(instances)
+
+		case <-launcher.connectionTimer.C:
+			log.Errorf("Nodes connection timeout")
+
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+func (launcher *Launcher) processRunInstanceStatus(runStatus NodeRunInstanceStatus) {
+	launcher.Lock()
+	defer launcher.Unlock()
+
+	log.Debugf("Received runstatus from nodeID: %s", runStatus.NodeID)
+
+	currentStatus := launcher.getNode(runStatus.NodeID)
+	if currentStatus == nil {
+		if !slices.Contains(launcher.config.SMController.NodeIDs, runStatus.NodeID) {
+			log.Errorf("Received status for unknown nodeID  %s", runStatus.NodeID)
+			return
+		}
+
+		var err error
+
+		currentStatus, err = launcher.initNodeStatus(runStatus.NodeID, runStatus.NodeType)
+		if err != nil {
+			log.Errorf("Can't init node: %v", err)
+
+			return
+		}
+
+		launcher.nodes = append(launcher.nodes, currentStatus)
+
+		if len(launcher.nodes) == len(launcher.config.SMController.NodeIDs) {
+			log.Debug("All clients connected")
+		}
+
+		slices.SortFunc(launcher.nodes, func(a, b *nodeStatus) bool {
+			if a.priority == b.priority {
+				return a.NodeID < b.NodeID
+			}
+
+			return a.priority < b.priority
+		})
+	}
+
+	currentStatus.receivedRunInstances = runStatus.Instances
+	currentStatus.currentRunRequest = nil
+	currentStatus.waitStatus = false
+
+	if len(launcher.nodes) != len(launcher.config.SMController.NodeIDs) {
+		return
+	}
+
+	for _, node := range launcher.nodes {
+		if node.waitStatus {
+			return
+		}
+	}
+
+	log.Info("All SM statuses received")
+
+	launcher.connectionTimer.Stop()
```

Comment:
Do we receive GetRunInstancesStatusChannel all the time ?  maybe change log above, I mean in first-time show this message log, but in next time show only `receive new statuses from SM`